docs/support/troubleshooting-failed-or-hanging-tasks.md

```diff
@@ -0,0 +1,55 @@
+
+If none of the above steps help, then you may have encountered a bug in a built-in step, in which case you can contact support for further assistance.
+
+## Networking Errors
+
+If your task logs contain errors that indicate a networking issue, there could be a few possible causes.
+
+### Connections between Octopus Server and Tentacles
+
+Octopus Server communicates with Tentacles in either Listening mode or Polling mode. Both modes require different configuration. 
+
+A common problem is that traffic on the appropriate ports (10933 by default for Listening Tentacles) is not allowed by your firewall. If you are encountering problems with your connections, then your Task log might show messages that indicate a connection timing out, or a connection that was rejected by the remote host.
+
+A utility called [TentaclePing](https://github.com/OctopusDeploy/TentaclePing) can be used to test and diagnose connections between the machine hosting Octopus Server and the machines hosting Tentacles. This allows you to quickly test connections in isolation, without involving the complexity of Octopus Server, Tentacles or tasks.
+
+See [Tentacle Communication Modes](docs/infrastructure/deployment-targets/windows-targets/tentacle-communication.md) for more information on configuring your Tentacles.
+
+#### Halibut
+
+The tool that manages connections between Octopus Server and your Tentacles is called [Halibut](https://github.com/OctopusDeploy/Halibut/). In order to discover more detailed information about the connections, it may be useful to [increase the log level for Halibut](docs/support/log-files.md#Logfiles-Changingloglevelshalibut). The same change to increase the log level for Halibut can also be made on [Tentacle](docs/support/log-files.md#Logfiles-Changingloglevelstentacle).
+
+### Connections to external services
+
+Steps that execute on Deployment Targets or Workers may need to reach out to contact other external services. In these cases, a useful first step to help diagnose the problem is to attempt to manually perform the same connection - this can help to isolate the problem to a networking issue rather than a problem with Octopus Deploy.
+
+Remember that these connections are usually initiated by your Deployment Targets or Workers, and not by Octopus Server. You may need to remotely connect to these remote machines, and then initiate a connection from those machines. This is distinct from attempting to establish the connection on the machine that hosts Octopus Server itself.
+
+## Hanging Tasks
+
+Sometimes tasks appear to be unresponsive or "hanging". The first step to diagnosing these problems is to examine your logs and determine the exact location that the task became unresponsive. If this occurs within the logs output by a custom script, then the bug is likely to originate from your script.
```

Comment:
I might be scarred here, but I would like to see a stronger "99/100 times it's anti-virus or anti-malware". The number one first step is to completely uninstall the AV software to eliminate it as the source of the problem.

Reply:
Good suggestion :+1: I'll tweak the wording